python2/typing.py

```diff
@@ -1857,6 +1858,26 @@ class ValuesView(MappingView[VT_co]):
     __extra__ = collections_abc.ValuesView
 
 
+class ContextManager(Generic[T_co]):
+    __slots__ = ()
+
+    def __enter__(self):
+        return self
+
+    @abc.abstractmethod
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Raise any exception triggered within the runtime context."""
+        return None
+
+    @classmethod
+    def __subclasshook__(cls, C):
+        if cls is ContextManager:
+            if (any("__enter__" in B.__dict__ for B in C.__mro__) and
+                    any("__exit__" in B.__dict__ for B in C.__mro__)):
```

Comment:
Changing this required more changes in the flake8 configuration than I would have liked. To keep this PR focused, maybe we should keep the cringeworthy formatting here for now, then change the flake8 configuration in a separate PR?